src/wp-includes/rest-api/endpoints/class-wp-rest-block-pattern-categories-controller.php

```diff
@@ -110,6 +145,11 @@ public function prepare_item_for_response( $item, $request ) {
 			}
 		}
 
+		// For backwards compatibility we only want to include the id if the field is explicitly requested.
+		if ( rest_is_field_included( 'id', $fields ) && isset( $item['id'] ) ) {
+			$data['id'] = $item['id'];
+		}
```

Comment:
Generally, introducing a new field would not be seen as a BC break. Is there a specific issue we are trying to work around here?

Reply:
ok, good to know, I thought additional fields would be seen as breaking change, I can remove this if not.